src/main/java/io/vertx/cassandra/CassandraClient.java

```diff
@@ -27,19 +27,70 @@
 import io.vertx.core.Vertx;
 
 import java.util.List;
+import java.util.UUID;
 
 /**
  * Eclipse Vert.x Cassandra client.
  */
 @VertxGen
 public interface CassandraClient {
 
-  static CassandraClient create(Vertx vertx) {
-    return create(vertx, new CassandraClientOptions());
+  /**
+   * The name of the default pool
+   */
+  String DEFAULT_POOL_NAME = "DEFAULT_POOL";
+
+
+  /**
+   * Like {@link CassandraClient#createNonShared(Vertx, CassandraClientOptions)}  but with default client options
+   */
+  static CassandraClient createNonShared(Vertx vertx) {
+    return createNonShared(vertx, new CassandraClientOptions());
+  }
+
+  /**
+   * Create a Cassandra client which maintains its own data source.
+   *
+   * @param vertx                  the Vert.x instance
+   * @param cassandraClientOptions the options
+   * @return the client
+   */
+  static CassandraClient createNonShared(Vertx vertx, CassandraClientOptions cassandraClientOptions) {
+    return new CassandraClientImpl(vertx, UUID.randomUUID().toString(), cassandraClientOptions);
   }
 
-  static CassandraClient create(Vertx vertx, CassandraClientOptions cassandraClientOptions) {
-    return new CassandraClientImpl(vertx, cassandraClientOptions);
+  /**
+   * Like {@link CassandraClient#createShared(Vertx, String, CassandraClientOptions)}, but with default client options and datasource
+   */
+  static CassandraClient createShared(Vertx vertx) {
+    return createShared(vertx, DEFAULT_POOL_NAME);
+  }
+
+  /**
+   * Like {@link CassandraClient#createShared(Vertx, String, CassandraClientOptions)}, but with default client options
+   */
+  static CassandraClient createShared(Vertx vertx, String datasourceName) {
+    return createShared(vertx, datasourceName, new CassandraClientOptions());
+  }
+
+  /**
+   * Like {@link CassandraClient#createShared(Vertx, String, CassandraClientOptions)}, but with datasource name
+   */
+  static CassandraClient createShared(Vertx vertx, CassandraClientOptions cassandraClientOptions) {
+    return createShared(vertx, DEFAULT_POOL_NAME, cassandraClientOptions);
+  }
+
+  /**
+   * Create a Cassandra client which shares its data source with any other Cassandra clients created with the same
+   * data source name
```

Comment:
Missing dot at the end of the first sentence for Javadoc to be valid.

Reply:
Same applies in other Javadocs.

Reply:
Done